sgpt/handlers/chat_handler.py

```diff
@@ -126,52 +122,45 @@ def show_messages(cls, chat_id: str) -> None:
             return
         # Prints all messages from a specified chat ID to the console.
         for index, message in enumerate(cls.chat_session.get_messages(chat_id)):
-            message = message.replace("\nCommand:", "").replace("\nCode:", "")
             color = "cyan" if index % 2 == 0 else "green"
             typer.secho(message, fg=color)
         raise typer.Exit()
 
-    def validate(self) -> None:
+    def validate(self) -> List[Mapping[dict, dict]]:
         if self.initiated:
-            if self.is_shell_chat and self.mode == CompletionModes.CODE:
-                raise BadArgumentUsage(
-                    f'Chat session "{self.chat_id}" was initiated as shell assistant, '
-                    "and can be used with --shell only"
-                )
-            if self.is_code_chat and self.mode == CompletionModes.SHELL:
+            # TODO: Make this system able to tell what role a chat was started with, validate,
+            #  and continue if the user didn't specify a role
+            existing_system_prompt = self.chat_history[0]["content"]
+            hypothetical_messages = make_prompt.prompt_constructor(
+                self.prompt, role=self.role, chat_init=False
+            )
+            if hypothetical_messages[0]["content"] == existing_system_prompt:
+                # return everything except the first message
+                # because the system message is already in the chat history,
+                # and chat_cache will add it again
+                messages = hypothetical_messages[1:]
+            else:
+                # Error message about how the user is asking for a different
+                # system chat than the one that already exists
                 raise BadArgumentUsage(
-                    f'Chat "{self.chat_id}" was initiated as code assistant, '
-                    "and can be used with --code only"
+                    f"Chat id:'{self.chat_id}' was initiated with system role as"
+                    f" \n'{existing_system_prompt}'\n\n "
+                    f"Can't be continued with new system"
+                    f" prompt \n'{hypothetical_messages[0]['content']}'"
                 )
```

Comment:
This is currently how I am handling the existing chat history being for a different role. It would be really nice to have some variable saved into the chat history so the user wasn't confronted with a wall of text whenever they enter the wrong role for a chat. 

This would also make it much easier to identify what chat was used, so potentially Shell GPT could automatically select the correct role. (Bearing in mind that role files may change.)